Improved error catching for getJSON calls

alex-deaconu opened the pull request:
Added a fail function for getJSON calls
Fail function logs any errors with the call

Unfortunately, *No Internet Connection* scenarios are not caught

Few other fixes/updates for logging functionality

alex-deaconu commented:
@tejohnso I believe this is what you had in mind as a potential fix for Issue #166 . Can you please review and let me know if acceptable? Thanks!

tejohnso commented on src-js/Viewer/Data/DisplayRegistrationController.js:
```diff
@@ -42,12 +42,12 @@ function _callUrlNative(url, mode, cb) {
   } catch (err) {
-    Logger.log(`Register Display request error - ${url} - ${err.message}`);
+    Logger.log(`Register Display request error - ${url}`, err);
   }
```
@alex-deaconu I think this catch clause needs a similar treatment to the one in ViewerDataProvider.  Otherwise, can you clarify what errors this is catching?

alex-deaconu commented:
@tejohnso you're right. Looks like I missed that and a few other jQuery requests. See latest commit, I think I got all of them this time. Thanks!

tejohnso commented:
:+1:

alex-deaconu commented:
I've revised the code, after some testing. To summarize:
The `ajax` `complete` function gets called on both failure and success. Removed the extra `fail` function and moved success logging.
Also, found the `fail` `error` parameter to be unreliable. Updated with `textStatus` for some cases.

@tejohnso one more commit to review. Thanks.

tejohnso commented:
:+1:

alex-deaconu commented:
@tejohnso thanks!
